plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeFileOperations.java

```diff
@@ -742,12 +769,12 @@ private record FileOperation(FileType fileType, String fileId, OperationType ope
     {
         public static FileOperation create(String path, OperationType operationType)
         {
-            Pattern dataFilePattern = Pattern.compile(".*?/(?<partition>key=[^/]*/)?(?<queryId>\\d{8}_\\d{6}_\\d{5}_\\w{5})_(?<uuid>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})");
+            Pattern dataFilePattern = Pattern.compile(".*?/(?<partition>key=[^/]*/)?((?<queryId>\\d{8}_\\d{6}_\\d{5}_\\w{5})_(?<uuid>[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})|.*.parquet)");
```

Comment:
The `.*.parquet` is pretty broad.
For example, checkpoint files would also match the pattern.

Maybe instead of this,
let's turn `dataFilePattern` into an ordinary catch all pattern `\Q table_directory \E / ( key=value /)?  [^/]+`

Reply:
Modified the pattern to `.*?/(?<partition>key=[^/]*/)?[^/]+` and placed it only after matching against metadata files.